### ISSUE TITLE
Duplicate recordings

### ISSUE THREAD

**Issue by Gimpoid** [beaston02/MFCRecorder#11]
Just wondering if anyone has the same issue or has any idea how to fix this. I noticed far larger space usage than usual (30+GB). When I checked I found it had made 44 duplicate files for one model. What was also strange is they were all slightly different sizes and lengths. As time went on they seemed to be getting shorter and shorter.

They were all also created over a two hour period. I highlighted the duplicates in the following screenshot and you should be able to see the timestamps.

2:36:25 was the length of what i assume is the original and the last file was 1:50:08.

[http://imgur.com/a/yGqmI](url)

**Comment by beaston02** [comment]
I actually noticed this in testing once, and completely forgot to fix it.

Thank you for the reminder. Ill see about fixing this soon.

**Comment by Gimpoid** [comment]
No problem I think it has happened in the past I just never looked closely at the output as I was messing around with settings. I use it on a 100gb virtual drive and noticed I had somehow burnt 70gb in one night when it normally is only maybe 10-20 max. I had a look and found it occured with another model.

[http://imgur.com/a/aiXHD](url)

Let me know if there is anything I can do to help.

**Comment by beaston02** [comment]
Thanks again. 

Just to let you know, youre linking the images wrong. When you click the link icon, you get "\[](url)" and you need to replace the "url" with the link. whatever is between the "[]" will be the text shown, and the "(url)" is the actual link. to link the image from your last post it would be "\[whatever text you want displayed goes here](http://imgur.com/a/aiXHD )" 

which then is a clickable link to the image.
[whatever text you want displayed goes here](http://imgur.com/a/aiXHD)

**Comment by Gimpoid** [comment]
Thanks I didnt realise haha =)

I am not sure if this error is related but I just saw it now and did notice a few models have a ton of duplicates again today.

[Exceptions.txt](https://github.com/beaston02/MFCRecorder/files/1259312/Exceptions.txt)

**Comment by beaston02** [comment]
Thank you for including the exceptions. 

Off topic question. Does it always print the "...models recording next check in x seconds" to a new line? On unix systems, everything is logged to the same line (other than when exceptions occur). I didnt realize I needed to clean that up for windows. Im still trying to figure out a cleaner output in general as Im not really happy with how Im currently doing it. 

And to get back on topic, I think i fixed it in the update I just posted. I made the changes last night and ran it throughout the night with no issues, but maybe that was just luck! Please update and let me know if you continue to have the error.

**Comment by dannyvoid** [comment]
Yes, it outputs each check to a new line, every time.

**Comment by newtonslaw** [comment]
Hey,

New user here, and am seeing something interesting that looks like its related to this bug.  I have two models currently recording, one I have a single file for, so all good there.  The other I have over 100 for.  Interestingly they are not duplicates, they are overlapping.  So file 1 has 100% of the stream, file 2 starts say 20 seconds after, file 3 starts 20 seconds after that and so on.  20 seconds is not the exact number just an indicator.  There are no exceptions being logged, so I have nothing to share there I am afraid.  I do know that the model is having connectivity issues from talking to her.  Sorry this is a bit vague but all I have to give at the moment.

**Comment by beaston02** [comment]
@newtonslaw, I received an email alert with a comment from you on this topic, but it is not showing up here. That comment was:

> Hey,

> New user here, and am seeing something interesting that looks like its related to this bug. I have two models currently recording, one I have a single file for, so all good there. The other I have over 100 for. Interestingly they are not duplicates, they are overlapping. So file 1 has 100% of the stream, file 2 starts say 20 seconds after, file 3 starts 20 seconds after that and so on. 20 seconds is not the exact number just an indicator. There are no exceptions being logged, so I have nothing to share there I am afraid. I do know that the model is having connectivity issues from talking to her. Sorry this is a bit vague but all I have to give at the moment.


The issue you're seeing is the exact same issue as here. Each time a check was done, the model was started again although the previous recording was still running, resulting in 1 recording of the complete show, and a new one starting x seconds later, and therefor missing the that much more from the beginning of the session each time. The model was dropping off of the recording list while it was still attempting to pull from the live stream causing this issue anytime there was a "hiccup" in the connection to the stream. I posted an update yesterday (about 20 hours ago) which should have addressed this issue. I just posted another update and increased the attempts before dropping the model from the recording list from 3 to 5. It will have to have 5 attempts to pull data failing before the model is dropped now, which will hopefully be enough to prevent breaking a single session into multiple files, and it should eliminate the issue with restarting while still recording. 

@dannyvoid, Thank you for chiming in! I included what will hopefully fix the output in the latest update. I dont have access to a windows machine so I was unable to test this. Please let me know if it helps or not.

Also, I added a gevent timeout to (hopefully) test issue #8 (hanging on connection attempt) so it now requires the gevent library be installed (pip3 install gevent).

**Comment by newtonslaw** [comment]
Thanks @beaston02 

I deleted my comment as I believed I had actually made a mistake in my environment.  I was running under screen and was seeing this issue, I needed to do something else on the server that required me to reboot, after I rebooted my server the issue was gone, so I deleted my comment rather than have you looking for something that was not there.  Either there was an issue with my setup, or coincidentally something was fixed elsewhere at the same time.  Sorry about wasting your time not my intent. Thank for the update, I will apply it later on, and let you know how it goes.

**Comment by beaston02** [comment]
You werent wasting my time at all! I wondered if you deleted the comment. Im fairly new here myself so not exactly sure how it all works myself.

What you saw was most likely an issue in the script. Make sure you are running the latest update, and it should (hopefully) prevent the issue from happening again. It was an intermittent problem, so something you shouldnt have seen often before. MFC seemed to be doing some maintenance, or having issues the past couple of days, which most likely caused this issue to occur more frequently than usual, but the issue was still there in my code. 

Keep me updated on if you see the issue again or not! 

Ill leave this issue open for now, although i do believe it should now be fixed.

**Comment by Gimpoid** [comment]
Ill test it out tonight. Seems much nicer in windows without it constantly scrolling.

One thing to note is on windows your requirements.txt file appears as "geventrequests
livestreamer" with no line breaks so I had to manually add them before doing the install.

**Comment by beaston02** [comment]
>Seems much nicer in windows without it constantly scrolling.

Glad that worked! Its still not great, If enough models are recording, it will scroll to the next line mucking things up, and like I mentioned, errors will throw it off some too. Ill figure out a better solution later.
 
>One thing to note is on windows your requirements.txt file appears as "geventrequests
livestreamer" with no line breaks so I had to manually add them before doing the install.

Thats odd. I added an empty line between each. Hopefully that helps.

**Comment by Gimpoid** [comment]
Strange the requirements is still a single line in windows.

[requirements.txt](https://github.com/beaston02/MFCRecorder/files/1265317/requirements.txt)

There is my edited copy if that helps with linebreaks.

I had some errors last night but the recording looks to have continued past it rather than hanging so I am not sure what broke maybe just a connection error. I didn't see any duplicate files this morning which is good.

[New Text Document.txt](https://github.com/beaston02/MFCRecorder/files/1265319/New.Text.Document.txt)

**Comment by beaston02** [comment]
It seems MFC was having some issues yesterday. What you saw may be related to those issues. I noticed I kept getting URL errors, and I changed from urllib to requests in mfcauto.py and it instantly cleared it all up. If I ever get the connection issue fixed, Ill post my mfcauto version on here. I really have no clue whats going on with it. Everything I try makes no difference at all from what I can tell. Everything Ive been able to find on google also does not seem to make a difference. I suppose its one of the issues that comes with still being fairly new to it all, and part of the learning process :) 

Ill post your requirements.txt file here and hopefully that fixes it. for what its worth, on my mac it is each 2 blank lines between each module. I never thought windows and *nix systems would read .txt files so differently.
